plugins/node/opentelemetry-instrumentation-cassandra/src/types.ts

```diff
@@ -15,6 +15,16 @@
  */
 
 import { InstrumentationConfig } from '@opentelemetry/instrumentation';
+import { Span } from '@opentelemetry/api';
+import type * as CassandraDriver from 'cassandra-driver';
+
+export interface ResponseInfo {
+  response: CassandraDriver.types.ResultSet;
+}
```

Comment:
That's correct. We should not use anything from instrumented package in the public interface.

Reply:
That's true, this will throw as it's in the devDependency. But I'm not sure what the right approach is, should I move it to the regular dependency, or replace the type with `any` or a different object?

Reply:
yeah that's a bummer.

I can think of these 2 non-optimal options:
1. use `any` with a comment that points to the actual type to guide users on how to use this hook.
2. Copy the `ResultSet` type, along with it's dependencies into this file, with a comment that points to the source.

@Rauno56 - Do you have any other ideas?

Reply:
If I remember correctly, I've seen the 2nd option in other instrumentations, which I think is better than using `any` since it can be a bit confusing

Reply:
I agree, but in this specific case, [the `ResultSet` interface is quite complex and has transitive dependencies on other types](https://github.com/datastax/nodejs-driver/blob/d42176e4baa1cfc3df79699cc3b5d575c86e3cec/lib/types/index.d.ts#L323):
```js
 interface ResultSet extends Iterable<Row>, AsyncIterable<Row> {
    info: {
      queriedHost: string,
      triedHosts: { [key: string]: any; },
      speculativeExecutions: number,
      achievedConsistency: consistencies,
      traceId: Uuid,
      warnings: string[],
      customPayload: any
    };

    columns: Array<{ name: string, type: { code: dataTypes, info: any } }>;
    nextPage: (() => void) | null;
    pageState: string;
    rowLength: number;
    rows: Row[];

    first(): Row;

    wasApplied(): boolean;
  }
```

Maybe not all of them are relevant for the hook and we can just declare an interface with a subset?
I guess you are trying to collect the response payload which only cares about the `rows` property.

Reply:
True it has a lot of dependencies.
I feel like the user will expect the entire result to be in the hook since this is the expected behavior.
But I do agree with you that a subset could be sufficient since the most important part of the result is the `rows` property. Do you think we should go with that?